File: fix_function_declarations.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_function_declarations(file_path):
    """Fix missing function declarations in React components"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has React imports but no function declaration
        if 'import React' in content and 'export default' in content:
            lines = content.split('\n')
            new_lines = []
            has_function_declaration = False
            
            for i, line in enumerate(lines):
                # Check if this is the first line after imports that should be a function declaration
                if (line.strip().startswith('const ') or 
                    line.strip().startswith('let ') or 
                    line.strip().startswith('var ') or
                    line.strip().startswith('interface ') or
                    line.strip().startswith('type ') or
                    line.strip().startswith('const ') or
                    line.strip().startswith('  const ') or
                    line.strip().startswith('  let ') or
                    line.strip().startswith('  var ')):
                    
                    # Check if this is likely a component (has JSX or React patterns)
                    if any(keyword in line for keyword in ['useState', 'useEffect', 'React.FC', '= () =>', '= () => {']):
                        # Add function declaration before this line
                        if not has_function_declaration:
                            # Extract component name from export default line
                            export_line = None
                            for j in range(i, len(lines)):
                                if 'export default' in lines[j]:
                                    export_line = lines[j]
                                    break
                            
                            if export_line:
                                component_name = export_line.split()[-1]
                                new_lines.append(f'export default function {component_name}() {{')
                                has_function_declaration = True
                            else:
                                new_lines.append('export default function Component() {')
                                has_function_declaration = True
                    
                    new_lines.append(line)
                else:
                    new_lines.append(line)
            
            # If we added a function declaration, we need to close it
            if has_function_declaration and not content.strip().endswith('}'):
                new_lines.append('}')
            
            # Write the cleaned content
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write('\n'.join(new_lines))
            
            print(f"Fixed function declaration in: {file_path}")
            return True
        
        return False
        
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    # Find all TypeScript/TSX files
    pattern = "/workspace/**/*.tsx"
    files = glob.glob(pattern, recursive=True)
    
    fixed_count = 0
    for file_path in files:
        if fix_function_declarations(file_path):
            fixed_count += 1
    
    print(f"Fixed function declarations in {fixed_count} files")

if __name__ == "__main__":
    main()

File: fix-function-declarations.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const glob = require('glob');

// Function to fix function declarations
function fixFunctionDeclarations(content) {
  let fixed = content;

  // Fix malformed function declarations
  fixed = fixed.replace(/export default function ServicePage\(\) \{ return \(/g, 
    'export default function ServicePage() {\n  return (');
  
  // Fix malformed function declarations with different names
  fixed = fixed.replace(/export default function (\w+)\(\) \{ return \(/g, 
    'export default function $1() {\n  return (');
  
  // Fix any remaining malformed function declarations
  fixed = fixed.replace(/function (\w+)\(\) \{ return \(/g, 
    'function $1() {\n  return (');
  
  // Fix const function declarations
  fixed = fixed.replace(/const (\w+) = \(\) => \{ return \(/g, 
    'const $1 = () => {\n  return (');
  
  // Fix arrow function declarations
  fixed = fixed.replace(/const (\w+) = \(\) => \(/g, 
    'const $1 = () => (');
  
  // Fix malformed JSX returns
  fixed = fixed.replace(/return \(\s*<>\s*$/gm, 'return (\n    <>');
  
  // Fix closing braces and parentheses
  fixed = fixed.replace(/<\/>\s*\)\s*$/gm, '    </>\n  );');
  
  // Fix any remaining malformed returns
  fixed = fixed.replace(/return \(\s*$/gm, 'return (');
  
  return fixed;
}

// Function to process a single file
function processFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    const fixed = fixFunctionDeclarations(content);
    
    if (content !== fixed) {
      fs.writeFileSync(filePath, fixed, 'utf8');
      console.log(`Fixed function declarations in: ${filePath}`);
      return true;
    }
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Main function
function main() {
  const patterns = [
    'app/**/*.tsx',
    'app/**/*.ts',
    'components/**/*.tsx',
    'components/**/*.ts',
    'src/**/*.tsx',
    'src/**/*.ts'
  ];

  let totalFixed = 0;
  let totalProcessed = 0;

  patterns.forEach(pattern => {
    const files = glob.sync(pattern, { cwd: process.cwd() });
    
    files.forEach(file => {
      totalProcessed++;
      if (processFile(file)) {
        totalFixed++;
      }
    });
  });

  console.log(`\nProcessed ${totalProcessed} files`);
  console.log(`Fixed function declarations in ${totalFixed} files`);
}

if (require.main === module) {
  main();
}

module.exports = { fixFunctionDeclarations, processFile };